### ISSUE TITLE
Label LineBreakMode="CharacterWrap" doesn't work in android

### ISSUE THREAD

**Issue by sady4850** [xamarin/Xamarin.Forms#7118]
### Description

`Xararin.Forms.Label.LineBreakMode` `CharacterWrap` option works like WordWrap in android. No `Grid` and `StackLayout`, just Label inside ContentPage. 

### Steps to Reproduce

```xml
<?xml version="1.0" encoding="utf-8"?>
<ContentPage  xmlns="http://xamarin.com/schemas/2014/forms" xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml" xmlns:d="http://xamarin.com/schemas/2014/forms/design" xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" mc:Ignorable="d" x:Class="TestWrap.MainPage">
    <Label BackgroundColor="White" LineBreakMode="CharacterWrap" Text="WelcometoXamarinFor mssdgjhxdghxsdekghs ghsksdgkhdghdkhglsdghdhgkd hdsgdklhgkhdglhdfklhgdh" />
</ContentPage>
```

### Basic Information

- Version with issue:
Xamarin.Forms 4.1.0555618

### Screenshots

Android
<img width="383" alt="androidCharacterWrap" src="https://user-images.githubusercontent.com/13216814/62769125-0c472180-baa1-11e9-92d9-8c91483bd062.png">

iOS
<img width="435" alt="iOScharacterWrap" src="https://user-images.githubusercontent.com/13216814/62769142-1537f300-baa1-11e9-8266-3c44abaf9cae.png">

**Comment by jfversluis** [comment]
Created a little sample project to show this behavior. Indeed it doesn't seem to work like you'd expect.

Android does not seem to have this functionality built-in.

[Repro7118.zip](https://github.com/xamarin/Xamarin.Forms/files/3485812/Repro7118.zip)

**Comment by sady4850** [comment]
>Android does not seem to have this functionality built-in.

Indeed. People [suggest](https://stackoverflow.com/questions/5118367/android-how-to-wrap-text-by-chars-not-by-words) to replace spaces with no-break space character to achieve this.

**Comment by zoli13** [comment]
Attaching a sample project: [App1.zip](https://github.com/xamarin/Xamarin.Forms/files/5261042/App1.zip)

**iOS (proper) screenshot:**
![ios](https://user-images.githubusercontent.com/9251057/93876520-48563980-fcd7-11ea-83cf-61c521587150.png)

**Android (incorrect, see some are wrapped into 3 lines, while obviously 2 lines is enough, just as on iOS):**
![android](https://user-images.githubusercontent.com/9251057/93876591-62901780-fcd7-11ea-9e01-5a485e3877d9.png)
